mobile/src/bundles/talent/store/slice.ts

```diff
@@ -2,52 +2,63 @@ import { createSlice } from '@reduxjs/toolkit';
 
 import { DataStatus } from '~/bundles/common/enums/enums';
 import { type ValueOf } from '~/bundles/common/types/types';
-import { type ProfileStepDto } from '~/bundles/talent/types/types';
+import { type UserDetailsResponseDto } from '~/bundles/talent/types/types';
 
-import { completeProfileStep } from './actions';
+import {
+    createTalentDetails,
+    getTalentDetails,
+    setCompletedStep,
+    updateOnboardingData,
+} from './actions';
 
 type State = {
     dataStatus: ValueOf<typeof DataStatus>;
-    profileStepData: ProfileStepDto | null;
+    onboardingData: UserDetailsResponseDto | null;
+    completedStep: number;
 };
```

Comment:
I wanted to make a state 
```
dataStatus: ValueOf<typeof DataStatus>;
completedStep: number;
profileStepData: ProfileStepDto | null;
cvAndContactsStepData: CVAndContactsStepDto | null;
...
```
But I couldn't :(

Since POST and PATCH requests always return a complete object with all user-details fields, I left only onboardingData. And in each step, we need to get data according to its type. I don't know if it's okay.

Reply:
It's ok